netstandard/pkg/targets/netstandard/NETStandard.Library.targets

```diff
@@ -1,12 +1,29 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <NETStandardLibraryPackageVersion>#VERSION#</NETStandardLibraryPackageVersion>
+  </PropertyGroup>
+
   <!-- Only add references if we're actually targeting .NETStandard.
        If the project is targeting some other TFM that is compatible with NETStandard we expect
        that framework to provide all references for NETStandard, mscorlib, System.* in their own
        targeting pack / SDK. -->
-  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETStandard'">
-    <_NetStandardLibraryRefPath>$(MSBuildThisFileDirectory)\ref\</_NetStandardLibraryRefPath>
-  </PropertyGroup>
-
-  <!-- import the TFM-agnostic targets -->
-  <Import Project="..\$(MSBuildThisFile)"/>
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETStandard'">
+    <Reference Include="$(MSBuildThisFileDirectory)\ref\netstandard.dll">
+      <!-- Private = false to make these reference only -->
+      <Private>false</Private>
+      <!-- hide these from Assemblies view in Solution Explorer, they will be shown under packages -->
+      <Visible>false</Visible>
+      <NuGetPackageId>NETStandard.Library</NuGetPackageId>
+      <NuGetPackageVersion>$(NETStandardLibraryPackageVersion)</NuGetPackageVersion>    
+    </Reference>
+    <Reference Include="#NETSTANDARDFACADES#">
```

Comment:
That is going to turn into a pretty long include string. Are there any limits in the length of that over just having one item per facade?

Reply:
Not that I am aware of.  MSBuild is using managed code to parse this thing so its not like we impose any limits on strings in a file.  I'm trying to keep the overall text size down as well because I don't want to increase the file size/parsing by a ton.